### ISSUE TITLE
'ref var' incorrectly infers a nullable type

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#42018]
**Version Used**: 3d7c4d4

**Steps to Reproduce**:

```csharp
#nullable enable

using System;

public class C {
    string _field = "";
    
    public void M() {
        ref var local = ref _field;
        local = null;

        local = ref _field;
    }
}
```

**Expected Behavior**:

A warning is reported on `local = null`:

> warning CS8625: Cannot convert null literal to non-nullable reference type.

**Actual Behavior**:

A warning is reported on `local = ref _field`:

> warning CS8619: Nullability of reference types in value of type 'string' doesn't match target type 'string?'.

**Comment by RikkiGibson** [comment]
Related to #40477

@jcouv it feels like a 'ref var' needs to infer a non-nullable type when the referenced variable is of a non-nullable reference type.

**Comment by sharwell** [comment]
Note that `ref string? local` will produce a warning as well, so we are inferring a type that cannot be written as an explicit type without a warning.
